merino-admin/suggestion_providers/models.py

```diff
@@ -281,6 +281,41 @@ def get_type(self):
         return "keyword_filter"
 
 
+class ClientVariantSwitchConfig(ProviderConfig):
+    help_text = """
+        Provider switches between two providers based on whether a request's 
+        client_variants matches the configured client_variant string. If there 
+        is a match, suggestions will be given from the matching provider. If not,
+        the default provider will be used.
+    """
+    client_variant: models.CharField = models.CharField(
+        max_length=256,
+        help_text="""
+            If this string is found in a client_variants, the matching_provider 
+            will be used for suggestions. If not, the default_provider will be 
+            used.
+        """,
+    )
+    matching_provider: models.ForeignKey = models.ForeignKey(
+        ProviderConfig,
+        on_delete=models.SET_NULL,
+        null=True,
+        related_name="client_variant_switch_matching_provider",
+        help_text=INNER_PROVIDER_HELP_TEXT,
```

Comment:
It might be good to use a specific help text or add something to the constant provider help text to indicate that this provider is used specifically in the matching case. Same with the default one.